include/engine/guidance/route_step.hpp

```diff
@@ -162,6 +163,8 @@ inline RouteStep &RouteStep::ElongateBy(const RouteStep &following_step)
                          following_step.intersections.begin(),
                          following_step.intersections.end());
 
+    is_segregated = false;
```

Comment:
what about multi-step intersections. Are we ensuring that all scenarios will be processed in a single step? Shouldn't this be transferring the `is_segregated` flag from the following step?

Reply:
Yes, this logic will be more complex in future.